### PR TITLE
Support Elasticsearch 7

### LINKED ISSUE #234: Support Elasticsearch 7.x
Sentry Issue: [NATLAS-SERVER-5](https://sentry.technowizardry.net/sentry/natlas-server/issues/70/?referrer=github_integration)

Time to upgrade to Elasticsearch 7.x.

To upgrade, we need:
* Update mappings
* Update docker-compose
* Update elasticsearch-py client
* Update documentation

```
RequestError: RequestError(400, 'illegal_argument_exception', 'The mapping definition cannot be nested under a type [_doc] unless include_type_name is set to true.')
(5 additional frame(s) were not displayed)
...
  File "elasticsearch/client/utils.py", line 76, in _wrapped
    return func(*args, params=params, **kwargs)
  File "elasticsearch/client/indices.py", line 88, in create
    params=params, body=body)
  File "elasticsearch/transport.py", line 318, in perform_request
    status, headers_response, data = connection.perform_request(method, url, params, body, headers=headers, ignore=ignore, timeout=timeout)
  File "elasticsearch/connection/http_urllib3.py", line 186, in perform_request
    self._raise_error(response.status, raw_data)
  File "elasticsearch/connection/base.py", line 125, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
```

### PR DESCRIPTION
**Pull request by m4rcu5**
This PR creates support for Elasticsearch 7 while preserving 6.x compatibility. It could close #234

### CODE REVIEW

**Comment by 0xdade** [comment]
Hey @m4rcu5, thanks for the PR! I've asked @ajacques to review it before acceptance to make sure we're all squared away.

**Comment by ajacques** [comment]
Hi m4rcu5, thanks for your PR. This is something I've been wanting to do because I run ES 7, but I haven't gotten a chance to make the change.

I'll take a look at this in the next few days and get back to you with any feedback as soon as I can.

File: natlas-server/app/elastic/client.py [hunk 1]
```diff
@@ -38,7 +43,16 @@ def _initialize_indices(self):
...
 		for index in self.natlasIndices:
 			if not self.es.indices.exists(index):
-				self.es.indices.create(index, body=self.mapping)
+				self.es.indices.create(index)
+
+		# Avoid a race condition
+		time.sleep(2)
+
+		for index in self.natlasIndices:
+			if int(self.esversion.split(".")[0]) >= 7:
```
[the remark this answers is not in the document]
Comment (m4rcu5) [reply natlas-server/app/elastic/client.py]: Thanks. I wasn't aware of that module!
I had to bump the requirement to `2.10.0` to get the compare/match functions.

File: natlas-server/app/elastic/client.py [hunk 2]
```diff
@@ -83,7 +97,7 @@ def collate_source(self, documents):
 	def execute_search(self, **kwargs):
 		''' Execute an arbitrary search.'''
 		with self._new_trace_span(operation='search', **kwargs) as span:
-			results = self._execute_raw_query(self.es.search, doc_type='_doc', **kwargs)
+			results = self._execute_raw_query(self.es.search, doc_type='_doc', rest_total_hits_as_int=True, **kwargs)
```
[the remark this answers is not in the document]
Comment (m4rcu5) [reply natlas-server/app/elastic/client.py:100]: When dropping any pre-7 support, you could rewrite the code that looks for a count to check the `value` attribute. As these are quite some places, this looked like the cleanest way to keep backwards compatibility, without rewriting all those snippets to check if its a **dict** or **int**.

**Comment by m4rcu5** [comment]
@ajacques thanks for the review.
The PR is updated based on your comments.

I've ran the docker containers against Elasticsearch 7.6.2 and Elasticsearch 6.8.2, both worked as expected.

File: natlas-server/app/elastic/client.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (0xdade) [reply natlas-server/app/elastic/client.py:101]: It'll make the most sense to drop ES 6.x support when we upgrade to natlas 0.7.x, which will indicate a backwards incompatible change. I haven't really roadmapped out the features for this yet, though.

**Pull request merged**